src/sql-parser/src/parser.rs

```diff
@@ -152,6 +152,7 @@ enum Precedence {
     Cmp,
     Like,
     Other,
+    PrefixBitwiseNot,
```

Comment:
Is the precedence different than `Other`?

Reply:
Actually according to [this table](https://www.postgresql.org/docs/7.2/sql-precedence.html) in postgres all bitwise operators should actually have precedence `Other`, so I think that you are right and we can remove this.

Reply:
Does that mean the comment in the Cockroach docs about prededence of `~` was wrong?

Reply:
This is from [the Cockroach compatibility page](https://www.cockroachlabs.com/docs/stable/postgresql-compatibility.html#precedence-of-unary):

> In PostgreSQL, the unary ~ (bitwise not) operator has a low precedence. For example, the following query is parsed as ~ (1 + 2) because ~ has a lower precedence than +:

This statement is line with [the table referenced above](https://www.postgresql.org/docs/7.2/sql-precedence.html) (as well as our precedence table) where only the binary `+` and `-` have higher precedence than `~`, but `~` has the same precedence as `&` and `|` (everything falls under `other`). 

Based on that information as expected the following equations hold in Postgres:

```
~x | y = (~x) | y
~x - y = ~(x - y)
``` 

The are tests marked with the headers

```
# precedence between '~' and ("|", "&")
```
and
```
# precedence between '~' and ('+', '-')
```

ensure that this is also the case in Materialize.